### PR TITLE
Fix/cook

### LINKED ISSUE #2115: Cooks not working
**Minecolonies version**

minecolonies-universal-1.12.2-0.8.7264.

**Expected behavior**

Cooks, to, well, actually cook and serve food

**Actual behavior**

Stand around and demand food and nothing else

**Steps to reproduce the problem**
This only happens when you don't have a full stack of food to give them. Partial stacks are ignored. Which can be a problem if you're just starting out and don't *have* a full stack of anything yet. I'n bit sure if the cook will eventually time out and serve what you give her, but the one I had certainly didnb't show any signs of doing so. Which is bad when you're trying to transition your first 4 colonists from shoving food in their pockets to going to the restaurant. 

(oh, complete side note. I have Harvestcraft as well, which changes the bread recipe. Will that screw up the cook and the baker, oe will giving them wheat still work?)

### LINKED ISSUE #2119: [BUG] The food disappears with the restaurant on a citizen with full inventory
### Minecolonies version

version 1.12.2-0.8.6996

### Expected behavior
The food should not go in void when the cook feed a villager with full inventory. 

But what is the good behavior ? 
- Cook is waiting for an empty slot? But the citizen will also wait for food and wait forever

### Actual behaviour

When citizen are fully inventory (like deliveryman) and he go to the restaurant, he consume all food (food disappear one by one)

### Steps to reproduce the problem

1. Build restaurant
2. fulfill inventory of a hungry Citizen
3. Watch the cook's inventory emptying
4. Profit

Thanks !

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #2115
Closes #2119

# Changes proposed in this pull request:
- Fixes recipes on client side
- Fixes cook to start working for real
- Fixes cook deleting food if worker has full inv.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingCook.java [hunk 1]
```diff
@@ -13,8 +13,10 @@
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
 import net.minecraft.nbt.NBTUtil;
+import net.minecraft.tileentity.TileEntity;
 import net.minecraft.tileentity.TileEntityFurnace;
 import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentTranslation;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingCook.java]: MINOR Remove this unused import 'net.minecraft.util.text.TextComponentTranslation'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 2]
```diff
@@ -178,7 +177,8 @@ protected AbstractEntityAIBasic(@NotNull final J job)
           /*
            * Reset to idle if no specific tool is needed.
            */
-          new AITarget(() -> {
+          new AITarget(() ->
+          {
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:181]: MINOR Remove useless curly braces around statement and then remove useless return keyword (sonar.java.source not set. Assuming 8 or greater.) [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1602)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 3]
```diff
@@ -157,7 +155,8 @@ protected AbstractEntityAIBasic(@NotNull final J job)
             If yes, transition to NEEDS_ITEM.
             and wait for new items.
            */
-          new AITarget(() -> {
+          new AITarget(() ->
+          {
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:159]: MINOR Remove useless curly braces around statement and then remove useless return keyword (sonar.java.source not set. Assuming 8 or greater.) [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1602)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 100 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 80 major
* ![MINOR][MINOR] 15 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuildingWorker.java#L89](https://github.com/Minecolonies/minecolonies/blob/403f389cae60703e417400d0fc0eafb0722ccdae/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java#L89): Remove this call from a constructor to the overridable "getBuildingLevel" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)
1. ![CRITICAL][CRITICAL] [ProviderHandler.java#L27](https://github.com/Minecolonies/minecolonies/blob/403f389cae60703e417400d0fc0eafb0722ccdae/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/ProviderHandler.java#L27): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [ProviderHandler.java#L117](https://github.com/Minecolonies/minecolonies/blob/403f389cae60703e417400d0fc0eafb0722ccdae/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/ProviderHandler.java#L117): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/403f389cae60703e417400d0fc0eafb0722ccdae/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/403f389cae60703e417400d0fc0eafb0722ccdae/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/403f389cae60703e417400d0fc0eafb0722ccdae/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/403f389cae60703e417400d0fc0eafb0722ccdae/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/403f389cae60703e417400d0fc0eafb0722ccdae/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [AbstractBuildingWorker.java#L590](https://github.com/Minecolonies/minecolonies/blob/403f389cae60703e417400d0fc0eafb0722ccdae/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java#L590): Reduce this class from 84 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/403f389cae60703e417400d0fc0eafb0722ccdae/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 99 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 80 major
* ![MINOR][MINOR] 14 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuildingWorker.java#L89](https://github.com/Minecolonies/minecolonies/blob/50a4cc7aa4aa5f03d1e6683c1551b9dfb8621571/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java#L89): Remove this call from a constructor to the overridable "getBuildingLevel" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)
1. ![CRITICAL][CRITICAL] [ProviderHandler.java#L27](https://github.com/Minecolonies/minecolonies/blob/50a4cc7aa4aa5f03d1e6683c1551b9dfb8621571/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/ProviderHandler.java#L27): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [ProviderHandler.java#L117](https://github.com/Minecolonies/minecolonies/blob/50a4cc7aa4aa5f03d1e6683c1551b9dfb8621571/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/ProviderHandler.java#L117): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/50a4cc7aa4aa5f03d1e6683c1551b9dfb8621571/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/50a4cc7aa4aa5f03d1e6683c1551b9dfb8621571/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/50a4cc7aa4aa5f03d1e6683c1551b9dfb8621571/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/50a4cc7aa4aa5f03d1e6683c1551b9dfb8621571/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/50a4cc7aa4aa5f03d1e6683c1551b9dfb8621571/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [AbstractBuildingWorker.java#L590](https://github.com/Minecolonies/minecolonies/blob/50a4cc7aa4aa5f03d1e6683c1551b9dfb8621571/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java#L590): Reduce this class from 84 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/50a4cc7aa4aa5f03d1e6683c1551b9dfb8621571/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java [hunk 4]
```diff
@@ -508,14 +508,24 @@ public void serializeToView(@NotNull final ByteBuf buf)
...
                 removeRecipe(token);
             }
-            ByteBufUtils.writeTag(buf, StandardFactoryController.getInstance().serialize(token));
+            else
+            {
+                storages.add(storage);
+            }
+        }
+
+        buf.writeInt(storages.size());
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java:525]: Why are we writing the entire recipe? Are we not supposed to use the central storage?

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 5]
```diff
@@ -509,15 +509,20 @@ private AIState lookForRequests()
...
+                        worker.getCitizenData().createRequestAsync(firstDeliverableRequest.getRequest());
                     }
                     else
                     {
-                        worker.getCitizenData().createRequestAsync(firstDeliverableRequest.getRequest());
+                        worker.getCitizenData().createRequest(firstDeliverableRequest.getRequest());
                     }
                 }
+
+                if(async)
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: No. This changes changes critical behaviour. You need to change the initial if switch to exclude asyncs. See the AITarget on how to check if any none async requests are open.

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 6]
```diff
@@ -509,15 +509,20 @@ private AIState lookForRequests()
                 {
                     //Seems like somebody else picked up our stack.
                     //Lets try this again.
-                    if (!async)
+                    if (async)
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:513]: ALSO wth: why this change?

**Review by marchermans — CHANGES_REQUESTED** [review]

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java (hunk 4 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java:525]: I also thought that, but else we'd have to send over 5000 recipes on login and then a new one each time one changes.

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java (hunk 6 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:513]: style =P

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java (hunk 5 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: kk

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 99 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 80 major
* ![MINOR][MINOR] 14 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuildingWorker.java#L89](https://github.com/Minecolonies/minecolonies/blob/32e987e8ab8eebf29fd6ec463309b05c58e2d8a8/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java#L89): Remove this call from a constructor to the overridable "getBuildingLevel" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)
1. ![CRITICAL][CRITICAL] [ProviderHandler.java#L27](https://github.com/Minecolonies/minecolonies/blob/32e987e8ab8eebf29fd6ec463309b05c58e2d8a8/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/ProviderHandler.java#L27): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [ProviderHandler.java#L117](https://github.com/Minecolonies/minecolonies/blob/32e987e8ab8eebf29fd6ec463309b05c58e2d8a8/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/ProviderHandler.java#L117): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/32e987e8ab8eebf29fd6ec463309b05c58e2d8a8/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/32e987e8ab8eebf29fd6ec463309b05c58e2d8a8/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/32e987e8ab8eebf29fd6ec463309b05c58e2d8a8/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/32e987e8ab8eebf29fd6ec463309b05c58e2d8a8/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/32e987e8ab8eebf29fd6ec463309b05c58e2d8a8/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [AbstractBuildingWorker.java#L590](https://github.com/Minecolonies/minecolonies/blob/32e987e8ab8eebf29fd6ec463309b05c58e2d8a8/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java#L590): Reduce this class from 84 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/32e987e8ab8eebf29fd6ec463309b05c58e2d8a8/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 7]
```diff
@@ -463,7 +463,8 @@ private void updateWorkerStatusFromRequests()
     @NotNull
     private AIState lookForRequests()
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:464]: MAJOR Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 99 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 80 major
* ![MINOR][MINOR] 14 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuildingWorker.java#L89](https://github.com/Minecolonies/minecolonies/blob/10c5a93d80bd09c7d2a5e9261a639f5b5d34049b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java#L89): Remove this call from a constructor to the overridable "getBuildingLevel" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)
1. ![CRITICAL][CRITICAL] [ProviderHandler.java#L27](https://github.com/Minecolonies/minecolonies/blob/10c5a93d80bd09c7d2a5e9261a639f5b5d34049b/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/ProviderHandler.java#L27): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [ProviderHandler.java#L117](https://github.com/Minecolonies/minecolonies/blob/10c5a93d80bd09c7d2a5e9261a639f5b5d34049b/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/ProviderHandler.java#L117): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/10c5a93d80bd09c7d2a5e9261a639f5b5d34049b/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/10c5a93d80bd09c7d2a5e9261a639f5b5d34049b/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/10c5a93d80bd09c7d2a5e9261a639f5b5d34049b/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/10c5a93d80bd09c7d2a5e9261a639f5b5d34049b/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/10c5a93d80bd09c7d2a5e9261a639f5b5d34049b/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [AbstractBuildingWorker.java#L590](https://github.com/Minecolonies/minecolonies/blob/10c5a93d80bd09c7d2a5e9261a639f5b5d34049b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java#L590): Reduce this class from 84 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/10c5a93d80bd09c7d2a5e9261a639f5b5d34049b/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**